utils/src/main/kotlin/Extensions.kt

```diff
@@ -182,6 +182,31 @@ fun File.searchUpwardsForSubdirectory(searchDirName: String): File? {
     return currentDir
 }
 
+/**
+ * Search [this] directory upwards towards the root until a file called [searchFileName] is found and return this file,
+ * or return null if no such file is found.
+ */
+fun File.searchUpwardsForFile(searchFileName: String, ignoreCase: Boolean = false): File? {
+    if (!isDirectory) return null
+
+    var currentDir: File? = absoluteFile
+    var currentFile = currentDir?.let { resolveFileCase(it, searchFileName, ignoreCase) }
+    while (currentDir != null && currentFile == null) {
+        currentDir = currentDir.parentFile ?: break
+        currentFile = resolveFileCase(currentDir, searchFileName, ignoreCase)
+    }
+
+    return currentFile
+}
+
+private fun resolveFileCase(dir: File, fileName: String, ignoreCase: Boolean): File? {
```

Comment:
Just a hint: Kotlin supports functions local to functions, i.e. you could put this above between line 189 and 190 (without the `private`).